README.md

```diff
@@ -24,66 +33,128 @@ var parser = require('postcss-value-parser');
+
+- **type**: The type of node (`word`, `string`, `div`, `space`, or `function`).
+  Each type is documented below.
+- **value**: Each node has a `value` property; but what exactly `value` means
+  is specific to the node type. Details are documented for each type below.
+- **sourceIndex**: The starting index of the node within the original source
+  string. For example, given the source string `10px 20px`, the `word` node
+  whose value is `20px` will have a `sourceIndex` of `5`.
+
+### word
+
+The catch-all node type that includes keywords (e.g. `no-repeat`),
+quantities (e.g. `20px`, `75%`, `1.5`), and hex colors (e.g. `#e6e6e6`).
+
+Node-specific properties:
 
-- `{ type: 'word', value: 'any' }`
-- `{ type: 'string', value: 'string', quote: '"' }`
-- `{ type: 'string', value: 'string', quote: '\'' }`
-- `{ type: 'div', value: '/' , before: ' ', after: ' ' }`
-- `{ type: 'div', value: ',', before: ' ', after: ' ' }`
-- `{ type: 'div', value: ':', before: ' ', after: ' ' }`
-- `{ type: 'space', value: ' ' }` space as a separator
-- `{ type: 'function', value: 'name', before: '', after: '', nodes: [] }`
+- **value**: The "word" itself.
+
+Unquoted `url()` arguments are also considered word nodes,
+e.g. `foo/bar.gif` in `url(foo/bar.gif)`.
+
+### string
+
+A quoted string value, e.g. `"something"` in `content: "something";`.
+
+Node-specific properties:
+
+- **value**: The text content of the string.
+- **quote**: The quotation mark surrounding the string, either `"` or `'`.
+
+### div
+
+A divider, for example
+
+- `,` in `animation-duration: 1s, 2s, 3s`
+- `/` in `border-radius: 10px / 23px`
+- `:` in `(min-width: 700px)`
+
+Node-specific properties:
+
+- **value**: The divider character. Either `,`, `/`, or `:` (see examples above).
+- **before**: Whitespace before the divider.
+- **after**: Whitespace after the divider.
+
+### space
+
+Whitespace used as a separator, e.g. ` ` occurring twice in `border: 1px solid black;`.
+
+Node-specific properties:
+
+- **value**: The whitespace itself.
+
+### function
+
+A CSS function, e.g. `rgb(0,0,0)` or `url(foo.bar)`.
+
+Function nodes have nodes nested within them: the function arguments.
+
+Additional properties:
+
+- **value**: The name of the function, e.g. `rgb` in `rgb(0,0,0)`.
+- **before**: Whitespace after the opening parenthesis and before the first argument,
+  e.g. `  ` in `rgb(  0,0,0)`.
+- **after**: Whitespace before the closing parenthesis and after the last argument,
+  e.g. `  ` in `rgb(0,0,0  )`.
+- **nodes**: More nodes representing the arguments to the function.
+
+Media features surrounded by parentheses are considered functions with an
+empty value. For example, `(min-width: 700px)` parses to these nodes:
+
+```js
+[
+  {
+    type: 'function', value: '', before: '', after: '',
+    nodes: [
+      { type: 'word', value: 'min-width' },
+      { type: 'div', value: ':', before: '', after: ' ' },
+      { type: 'word', value: '700px' }
+    ]
+  }
+]
+```
 
```

Comment:
Here it would be nice to show some examples of the parsing for `url` functions, as the behaviour is slightly different. Perhaps it might be nice to include this sentence, and an example with a quoted string and one without.

> Additionally, `url` functions are parsed differently depending on whether the first non-space character is a quote. `url("foo/bar/baz.jpg")` is parsed to:
>
> But `url(foo/bar/baz.jpg)` is parsed to: